### PR TITLE
Use protobuf's own cmake scripts and add travis for ios

### PR DESCRIPTION
**Pull request by Yangqing**

### CODE REVIEW

File: cmake/ProtoBuf.cmake [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
+if (ANDROID OR IOS)
+  option(protobuf_BUILD_SHARED_LIBS "" OFF)
+  option(protobuf_BUILD_TESTS "" OFF)
+  option(protobuf_BUILD_EXAMPLES "" OFF)
+  if (APPLE)
+    # Protobuf generated files triggers a deprecated atomic operation warning
+    # so we turn it off here.
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-declarations")
+  endif()
+  add_subdirectory(${CMAKE_SOURCE_DIR}/third_party/protobuf/cmake)
```
Comment (bwasti) [inline cmake/ProtoBuf.cmake]: ${PROJECT_SOURCE_DIR}

File: cmake/ProtoBuf.cmake [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
+  include_directories(SYSTEM ${CMAKE_SOURCE_DIR}/third_party/protobuf/src)
```
Comment (bwasti) [inline cmake/ProtoBuf.cmake]: ibid.

**Review by bwasti — CHANGES_REQUESTED** [review]

File: scripts/build_host_protoc.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# Build script to build the protoc compiler for the host platform.
+##############################################################################
+# This script builds the protoc compiler for the host platform, which is needed
+# for any cross-compilation as we will need to convert the protobuf source
+# files to cc files.
+#
+# After the execution of the file, one should be able to find the host protoc
+# binary at build_host_protoc/bin/protoc.
+
+CAFFE2_ROOT="$( cd "$(dirname "$0")"/.. ; pwd -P)"
```
[the remark this answers is not in the document]
Comment (Yangqing) [reply scripts/build_host_protoc.sh]: done

File: cmake/ProtoBuf.cmake (hunk 1 above)
Comment (Yangqing) [reply cmake/ProtoBuf.cmake]: done

File: cmake/ProtoBuf.cmake (hunk 2 above)
Comment (Yangqing) [reply cmake/ProtoBuf.cmake]: done

**Pull request closed without merging**